test/integration/signature_editor_spec.mjs

```diff
@@ -0,0 +1,221 @@
+/* Copyright 2025 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  cleanupEditing,
+  closePages,
+  getEditorSelector,
+  loadAndWait,
+  switchToEditor,
+  waitForTimeout,
+} from "./test_utils.mjs";
+
+const switchToSignature = switchToEditor.bind(null, "Signature");
+
+describe("Signature Editor", () => {
+  const descriptionInputSelector = "#addSignatureDescription > input";
+  const addButtonSelector = "#addSignatureAddButton";
+
+  describe("Basic operations", () => {
+    let pages;
+
+    beforeAll(async () => {
+      pages = await loadAndWait("empty.pdf", ".annotationEditorLayer", null, {
+        eventBusSetup: eventBus => {
+          eventBus.on("annotationeditoruimanager", ({ uiManager }) => {
+            window.uiManager = uiManager;
+          });
+        },
+      });
+    });
+
+    afterEach(async () => {
+      await cleanupEditing(pages, switchToSignature);
+    });
+
+    afterAll(async () => {
+      await closePages(pages);
+    });
```

Comment:
Given the current push (done by @timvandermeij) to get the integration-tests to run in a random order, would it perhaps be better to "properly" isolate each test instead?
```suggestion
    beforeEach(async () => {
      pages = await loadAndWait("empty.pdf", ".annotationEditorLayer", null, {
        eventBusSetup: eventBus => {
          eventBus.on("annotationeditoruimanager", ({ uiManager }) => {
            window.uiManager = uiManager;
          });
        },
      });
    });

    afterEach(async () => {
      await closePages(pages);
    });
```

Reply:
Yep it was the idea of having `await cleanupEditing(pages, switchToSignature);`  in the `afterEach` and this way it avoids to have to close the tab after each test.

Reply:
But that can still *potentially* leave other state around that might mess with future subsequent tests, and the only simple way to ensure that no "invisible" cross-dependencies exist between integration-tests is to run each one *truly* on its own (i.e. in its own tab).